### PR TITLE
CORTX-29807 Update README.md with granular data-only deployment

### PR DESCRIPTION
**Pull request by zahidsk**
Signed-off-by: Zahid Shaikh <zahid.shaikh@seagate.com>

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds new functionality)
- [ ] Breaking change (bug fix or new feature that breaks existing functionality)
- [ ] Third-party dependency update
- [x] Documentation additions or improvements
- [ ] Code quality improvements to existing code or test additions/updates

## Applicable issues

- This change fixes an issue: # CORTX-29807
- This change is related to an issue: # CORTX-29807

## CORTX image version requirements

This change requires the following images:

- `cortx-all:<version>`
- `cortx-rgw:<version>`

## Checklist

- [ ] The change is tested and works locally.
- [ ] New or changed settings in the solution YAML are documented clearly in the README.md file.
- [x] All commits are signed off and are in agreement with the [CORTX Community DCO and CLA policy](https://github.com/Seagate/cortx/blob/main/doc/dco_cla.md).

If this change addresses a CORTX Jira issue:

- [x] The title of the PR starts with the issue ID (e.g. `CORTX-XXXXX:`)

-----
[View rendered README.md](https://github.com/zahidsk/cortx-k8s/blob/CORTX-29807/README.md)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -178,6 +179,24 @@ If you have direct access to the underlying Kubernetes Nodes in your cluster, CO
 
 **_TODO_** Port https://seagate-systems.atlassian.net/wiki/spaces/PUB/pages/754155622/CORTX+Kubernetes+N-Pod+Deployment+and+Upgrade+Document+using+Services+Framework#5.-Understanding-Management-and-S3-Endpoints-and-configuring-External-Load-balancer-service(Optional) here or into a linked `doc/readme` file.
 
+### Granular Deployment (Data Pod Only)
+
+Apart from the steps given in [Deploying CORTX on Kubernetes](#deploying-cortx-on-kubernetes) steps above, add following steps for granular data Pod only deployment.
+    
+1.  Update the Deployment Mode in solution configuration file like solution.yaml: 
+By default, Deployment Mode is set to standard 
+For data pod only deployment set the deployment_type to data-only
```
Comment (walterlopatka) [inline README.md:188]:
```suggestion
for data pod only deployment set the deployment_type to data-only
```

File: README.md [hunk 2]
```diff
@@ -178,6 +179,24 @@ If you have direct access to the underlying Kubernetes Nodes in your cluster, CO
...
+    
+    ```bash
+    deployment_type: data-only
+    ```
+    
+   Currently framework support the following 2 modes:
+   - standard - Deploy all redefine pods
+   - data-only - Deploy only Data pod
+
+2.  Validate deployment, please note here you will have only Data Pod.
```
Comment (walterlopatka) [inline README.md:198]: Do we really need this line?  The main deploy documentation already describes how to validate the deployment.  Why add an additional step here?

**Review by walterlopatka — COMMENTED** [review]
Some general comments:

* I would avoid the term "granular deployment" in these instructions.  This is a term that we use to describe the general work of deploying a subset of CORTX pods.  I think the feature here is "Deploy cortx-data pods only."
* I would use the term "corx-data" pod rather than "Data Pods" because we have not defined Data Pods anywhere, and cortx-data seems like the closed thing to a technical name, since it is the base name of the pods.
* In the "Global Parameters" section there is already a reference to the `data-only` option.  I wonder if this new section is really helpful.  Maybe it would be better to describe that option more thoroughly?

To be clear, my main requests here are:
* Title the section "Deploy cortx-data- pods only"
* Consider if it would be more helpful to just describe "data-only" in the global parameters section below.  If we think it is, indeed, better to have this new section, consider adding a link from the Global Parameters table that mentions the data-only setting.

**Review by osowski — CHANGES_REQUESTED** [review]
I would expect the majority of this content to exist under the 3rd list item under https://github.com/Seagate/cortx-k8s/blob/39b8afc991701e172b61adf32e220a6a644c34b8/README.md#deploying-cortx-on-kubernetes, with a linked reference to the https://github.com/Seagate/cortx-k8s/blob/39b8afc991701e172b61adf32e220a6a644c34b8/README.md#global-parameters section below.

IMO this should not be its own stand-alone section under "## Getting Started".

I would add a second list item after "Update the namespace ..." with the commentary necessary for lines 186-188 here to have the same effect. IE "Update the deployment_type with the desired deployment mode. See below under Global Parameters for more details..."

File: README.md (hunk 2 above)
Comment (zahidsk) [reply README.md]: Removed 
I added this section to notify that in data only deployment we will see only data pod.
I think it is not required

File: README.md (hunk 1 above)
Comment (zahidsk) [reply README.md]: Review comments addressed

**Comment by zahidsk** [comment]
changes done as per review comments from @osowski

**Review by ujjwalpl — APPROVED** [review]

**Review by osowski — APPROVED** [review]
looks good to me!

**Review by walterlopatka — APPROVED** [review]

**Pull request merged**